### PR TITLE
Refactor horizon tutorial and rerun Geometry tutorial

### PR DESCRIPTION
**Pull request by HollowPrincess**
Updating the old tutorial: deleting unstructured horizon, restructuration, and updates, adding representation, cache details and additional manipulations with surfaces.

Also, rerun the SeismicGeometry tutorial with renewed cache API.

### CODE REVIEW

**Comment by akoryagin** [comment]
<p>+1, not clear. Yet, I don't think it is a <em>ratio. </em>It's more of a metric</p>

---

 <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/147/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by HollowPrincess** [comment]
<p><strong><em>Proposal:</em></strong></p><p> </p><p>1) Coverage <strong>metric </strong>shows <strong>the ratio of labeled non-dead traces to all non-dead traces</strong>: being more than <code>1</code> means that the horizon does not match the geology.</p><p> </p><p>2) Coverage <strong>metric </strong>shows <strong>the ratio of labeled non-dead traces</strong>: being more than <code>1</code> means that the horizon does not match the geology.</p>

---

 <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/147/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by HollowPrincess** [comment]
<p>Contents:</p><p> </p><p>* [Horizon initialization](#initialization) from a storage</p><p>* [Horizon representations](#representations) as matrices and points clouds</p><p>* [Geological attributes and basic data manipulations](#attributes) as amplitudes, phases, decompositions, surfaces quality evaluations, etc.</p><p>* [Caching](#cache) for faster repeated data loading</p><p>* [Additional manipulations](#add_manipulations) as creating carcasses, creating holes on horizons and cutting sharp discontinuites</p>

---

 <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/147/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by HollowPrincess** [comment]
<p>We can learn a lot by looking at geological transformations of the horizon depth map and the amplitudes along it. <strong>For example, visualizations of transformations can provide information about facies or possible geological fault locations.</strong>&nbsp;</p>

---

 <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/147/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by HollowPrincess** [comment]
<p> </p><p>The <code>evaluate</code> method prints several key characteristics <strong>and displays a map of support correlation. If traces correlate with supports, then they are green. The closer the color to red, the more complex a region (more it differs from reference traces).</strong></p>

---

 <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/147/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by HollowPrincess** [comment]
<p> </p><p>- load, inspect and visualize horizons by computing key characteristics (coverage, amount of labeled and non-labeled traces, etc.) and geological attributes (amplitudes, phases, decompositions)</p><p>- apply useful manipulations to horizons as cutting carcasses, holes, and sharp discontinuities</p><p>- cache horizons for saving time on repeated actions</p><p> </p>

---

 <a href='https://app.reviewnb.com/gazprom-neft/seismiqb/pull/147/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: seismiqb/src/utils/cache.py [hunk 1]
```diff
@@ -219,12 +222,14 @@ def get_cache_size(self, cached_methods=None):
...
 
+        # Accumulate cache size over all cached methods
+        # Each term is a size of cached numpy array
         for method in cached_methods:
             method_values = list(method.cache()[self].values())
 
             for values in method_values:
                 if isinstance(values, np.ndarray):
-                    cache_size_accumulator += sum(item.nbytes / (1024 ** 3) for item in method_values)
+                    cache_size_accumulator += sum(item.nbytes / (1024 ** 3) for item in values)
```
Comment (SergeyTsimfer) [inline seismiqb/src/utils/cache.py:232]: Why do you need to re-iterate over it?

File: seismiqb/src/utils/cache.py [hunk 2]
```diff
@@ -262,4 +273,23 @@ def cache_repr(self):
...
+
+            # Columns sort
+            if len(cache_repr_) > 0:
+                cache_repr_ = cache_repr_.loc[:, ['cache_length', 'cache_size', 'arguments']]
+
         return cache_repr_
+
+    @property
+    def cache_repr(self):
+        """ Pandas DataFrame of cache representation which consists of names, cache length
```
Comment (SergeyTsimfer) [inline seismiqb/src/utils/cache.py:288]: DataFrame with cache representation, that contains `names`, `cache_length` and `cache_size` for each cached method.

**Pull request merged**